operator/controllers/seldondeployment_controller.go

```diff
@@ -1518,13 +1518,20 @@ func (r *SeldonDeploymentReconciler) createDeployments(components *components, i
 						instance.Status.Replicas = found.Status.Replicas
 					}
 				}
-				log.Info("Deployment status ", "name", found.Name, "status", found.Status)
+				log.Info("Deployment status", "name", found.Name, "status", found.Status)
 				if found.Status.ReadyReplicas == 0 || found.Status.UnavailableReplicas > 0 {
 					if ready {
-						condition := getDeploymentCondition(found, appsv1.DeploymentAvailable)
-						log.Info("Updating condition for deployment", "name", found.Name, "condition", condition)
-						instance.Status.SetCondition(machinelearningv1.DeploymentsReady, condition)
+
+						availableCondition := getDeploymentCondition(found, appsv1.DeploymentAvailable)
+						log.Info("Updating availableCondition for deployment", "name", found.Name, "availableCondition", availableCondition)
+						instance.Status.SetCondition(machinelearningv1.DeploymentsReady, availableCondition)
 						log.Info("Inference status", "status", instance.Status)
+
+						progressingCondition := getDeploymentCondition(found, appsv1.DeploymentProgressing)
+						if progressingCondition.IsFalse() && availableCondition.IsFalse() {
+							log.Info("Deployment is not progressing, returning failed status", "name", found.Name)
+							return false, false, nil
+						}
```

Comment:
Another catch is that we should return from inside the loop as we are looping over all Deployments to reconcile so we do need to return after the whole loop is done. This is why the current one has a boolean that is returned at the end outside an error.

Reply:
Resolved